ngraph/core/CMakeLists.txt

```diff
@@ -33,8 +33,13 @@ source_group("include" FILES ${PUBLIC_HEADERS})
 
 configure_file(include/ngraph/version.in.hpp include/ngraph/version.hpp)
 
+if (NGRAPH_STATIC_LIBRARY)
+# Create static library
+    add_library(ngraph STATIC ${LIBRARY_SRC} ${PUBLIC_HEADERS})
+else()
 # Create shared library
-add_library(ngraph SHARED ${LIBRARY_SRC} ${PUBLIC_HEADERS})
+    add_library(ngraph SHARED ${LIBRARY_SRC} ${PUBLIC_HEADERS})
```

Comment:
I think we can just add:
```cmake
add_library(ngraph ${LIBRARY_SRC} ${PUBLIC_HEADERS})
```
By default it will be `SHARED`, but we pass `cmake -DBUILD_SHARED_LIBS=OFF` (see https://cmake.org/cmake/help/v3.0/variable/BUILD_SHARED_LIBS.html), cmake sets target type as `STATIC`

Reply:
Linked to ngraph/CMakeLists.txt conversation

Reply:
It seems that by default BUILD_SHARED_LIBS is OFF, i can change that

But some other of the libraries already rely on that flag (onnx, onnx_importer, onnx_proto). Should i put STATIC explicit to them ? otherwise they will end up being dynamic as well (unlike today)

Reply:
I tried to compile the master version passing -DBUILD_SHARED_LIBS=ON and it has side effects, for instance it compiles onnx as a dll and fails later because it's in no export target.

So i'm afraid the CMake process relies on the fact that BUILD_SHARED_LIBS=OFF even when building shared versions of openvino's libraries.

Reply:
Unfortunately ONNX is a third-party dependency. Can we force disable BUILD_SHARED_LIBS for some components (if we want to use this flag for static/dynamic build)?